components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ext/WebAppManifest.kt

```diff
@@ -14,8 +14,21 @@ import androidx.core.net.toUri
 import mozilla.components.browser.state.state.CustomTabConfig
 import mozilla.components.browser.state.state.ExternalAppType
 import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.concept.engine.manifest.WebAppManifest.Icon.Purpose
 import mozilla.components.support.utils.ColorUtils.isDark
 
+private const val MIN_INSTALLABLE_ICON_SIZE = 192
+
+/**
+ * Checks if the web app manifest can be used to create a shortcut icon.
+ */
+fun WebAppManifest.isInstallable() = icons.any { icon ->
```

Comment:
Nice clean up! Are there tests we need to move here for now from the previous one? If there aren't, are can we write some? 😛